service/src/main/java/uk/gov/hmcts/reform/fpl/service/removeorder/RemoveOrderService.java

```diff
@@ -63,6 +66,32 @@ public RemovableOrder getRemovedOrderByUUID(CaseData caseData, UUID removedOrder
             .orElseThrow(() -> new RemovableOrderNotFoundException(removedOrderId));
     }
 
+    public Optional<StandardDirectionOrder> getRemovedSDO(CaseData caseData, CaseData caseDataBefore) {
+        List<Element<StandardDirectionOrder>> hiddenSDOs = caseData.getHiddenStandardDirectionOrders();
+        List<Element<StandardDirectionOrder>> previousHiddenSDOs = caseDataBefore.getHiddenStandardDirectionOrders();
+
+        if (!Objects.equals(hiddenSDOs, previousHiddenSDOs)) {
+            return hiddenSDOs.stream()
+                .filter(removedSDO -> !previousHiddenSDOs.contains(removedSDO))
+                .findFirst()
+                .map(Element::getValue);
+        }
```

Comment:
I know we had issues making the method generic but could this section become generic and reusable between both methods?

Reply:
yeah, It can be. moved the common part to a method